src/main.rs

```diff
@@ -198,21 +193,21 @@ fn main() {
     }
 
     // Parse both input files.
-    let ast_base = parse_file(&matches.free[0], &lex_l_path1, &yacc_y_path1);
-    let ast_diff = parse_file(&matches.free[1], &lex_l_path2, &yacc_y_path2);
+    let ast_base = parse_file(&args.arg_base_file, &lex_l_path1, &yacc_y_path1);
+    let ast_diff = parse_file(&args.arg_diff_file, &lex_l_path2, &yacc_y_path2);
 
     // Dump ASTs to STDOUT, if requested.
-    if dump_ast {
-        println!("{}", ast_base);
-        println!("{}", ast_diff);
+    if args.flag_ast {
+        writeln!(&mut stdout(), "{}", ast_base).ok();
```

Comment:
Hmm, I personally prefer `println`, because that's what it's for. The actual question is, why doesn't Rust support printing to stderr in a nice way. Defining the macro isn't that [hard](https://stackoverflow.com/questions/27588416/how-to-send-output-to-stderr).